blogs/2019/11/26/custom-data-format.md

```diff
@@ -0,0 +1,94 @@
+---
+Order: 53
+TOCTitle: Custom Data Format
+PageTitle: "Custom Data Format: Evolving HTML and CSS Language Features"
+MetaDescription: "Custom Data Format: Evolving HTML and CSS Language Features"
+Date: 2020-01-08
+Author: Pine Wu
+MetaSocialImage:
+---
+
+# Custom Data Format: Evolving HTML and CSS Language Features
+
+January 8, 2020 by Pine Wu, [@octref](https://github.com/octref)
+
+The web evolves, so do its languages. New entities land in HTML and CSS; [Custom Elements](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements) and [Houdini](https://developer.mozilla.org/en-US/docs/Web/Houdini) allow users to extend HTML and CSS semantics; many developers today develop on new languages that resemble HTML and CSS. Although HTML and CSS see increasingly flexible usage, editor support for them lags behind.
+
+To modernize the HTML and CSS language support in Visual Studio Code, we designed the [Custom Data Format](https://github.com/microsoft/vscode-custom-data), a declarative JSON format for describing HTML and CSS entities. VS Code's HTML and CSS language servers can read data defined in this format and provide language support for the newly defined HTML and CSS entities.
+
+In this blog post, we explain the data format and how users and extension authors can take advantage of it.
+
+## Using Custom Data Format
+
+VS Code provides information about HTML/CSS entities in auto completion and hover information:
+
+![HTML and CSS Language Features](html-css-language-features.png)
+
+With the Custom Data Format, you can define new HTML/CSS entities and get auto completion, hover information and other language features. To get started, write a JSON file:
+
+```json
+{
+  "version": 1.1,
+  "tags": [
+    {
+      "name": "my-button",
+      "description": "My button. You should use it as in `<my-button type='alert'></mybutton>`.",
+      "references": [
+        {
+          "name": "Bootstrap buttons",
+          "url": "https://getbootstrap.com/docs/4.0/components/buttons/"
+        }
+      ]
+    }
+  ]
+}
+```
+
+Then configure `html.customData` to point to the data file:
+
+```json
+{
+  "html.customData": ["./html.html-data.json"]
+}
+```
```

Comment:
Please make more clear that this is the end user example. You only refer to the extension authors in the next section, but we miss the example, where an extension does nothing else than providing custom data. Then the transition from the end user example to the extension example could be something like: "Since not all users want to configure custom data themselves, you can also provide an extensions ..."